python/mxnet/rnn/rnn_cell.py

```diff
@@ -380,8 +380,9 @@ def __init__(self, num_hidden, prefix='lstm_', params=None, forget_bias=1.0):
         self._hB = self.params.get('h2h_bias')
 
     @property
-    def state_shape(self):
```

Comment:
Should I separate these two or merge into one property? 
Choice 1: `state_shape()` and `state_layout()`
Choice 2: `state_info()` only and the state_shape is defined based on the result of state_info